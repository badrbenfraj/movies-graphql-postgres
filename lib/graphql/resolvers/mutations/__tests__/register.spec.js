const register = require('../register')
// const { saltAndHashPassword } = require('../../../services/login')
const { validEmail } = require('../../../../utils/validation')

// jest.mock('../../../services/login', () => ({
//   saltAndHashPassword: jest.fn().mockReturnValue({
//     hashPassword: 'hashpasword',
//     salt: 'thisissalt',
//   }),
// }))

jest.mock('../../../../utils/validation', () => ({
  validEmail: jest.fn().mockReturnValue(true),
}))

describe('mutations/register', () => {
  let db

  beforeEach(() => {
    db = {
      query: jest.fn(),
    }
  })

  it('returns error if invalid email', async () => {
    try {
      await register(
        null,
        {
          input: { email: 'batmanwaynecorp.org', password: 'robin4ever', name: 'Batman' },
        },
        { db }
      )
    } catch (e) {
      expect(e.message).toMatch(/Invalid email/)
    }
  })

  it('returns error if to short password', async () => {
    try {
      await register(
        null,
        {
          input: { email: 'batman@waynecorp.org', password: 'robin', name: 'Batman' },
        },
        { db }
      )
    } catch (e) {
      expect(e.message).toMatch(/Passwords needs to be at least 8 characters/)
    }
  })
})
