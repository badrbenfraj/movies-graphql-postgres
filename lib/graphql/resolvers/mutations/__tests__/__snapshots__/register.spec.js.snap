// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations/register calls db to check if user exists 1`] = `"SELECT * FROM users WHERE email = 'batman@waynecorp.org'"`;

exports[`mutations/register calls db to insert new user 1`] = `
"
      INSERT INTO users (password, name, email, salt)
      VALUES ('hashpasword', 'Batman', 'batman@waynecorp.org', 'thisissalt')
    "
`;
